packages/server/lib/browsers/firefox.ts

```diff
@@ -288,6 +298,14 @@ const defaultPreferences = {
   'media.getusermedia.insecure.enabled': true,
 
   'marionette.log.level': launcherDebug.log.enabled ? 'Debug' : undefined,
+
+  // where to download files
+  // 0: desktop
+  // 1: default "Downloads" directory
+  // 2: directory specified with 'browser.download.dir' (set dynamically below)
+  'browser.download.folderList': 2,
+  // prevents the download prompt for the specified types of files
+  'browser.helperApps.neverAsk.saveToDisk': downloadMimeTypes,
```

Comment:
nice hack lol, sucks that this is the only way to do it... i even checked the source code, no way to specify *...

https://searchfox.org/mozilla-central/source/uriloader/exthandler/nsExternalHelperAppService.cpp#2611-2617

Reply:
i guess users can add their own extensions to this list with a plugin, so not too big a deal, but we should probably document that if you're downloading files with weird extensions (or no extensions) you may need to modify this pref...

Reply:
Good idea. I think the recipe will be a good place to demonstrate that. Though users shouldn't have to worry about extensions, since it works off of mime types, which I imagine it would be a bit less common for a user to have a unique one.

Reply:
ah i see what you mean, it's mime types, not extensions, that should help